input-profile.js

```diff
@@ -0,0 +1,194 @@
+import {Component, Type} from '@wonderlandengine/api';
+import {vec3, quat} from 'gl-matrix';
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    static Properties = {
+        controller: {type: Type.Object},
+        handednessIndex: {
+            type: Type.Enum,
+            values: ['left', 'right'],
+            default: 'left',
+        },
+        path: {
+            type: Type.String,
+            default:
+                'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@1.0.13/dist/profiles/',
+        },
+        customProfileFolder: {
+            type: Type.String,
+        },
+    };
+
+    init() {
+        this.gamepadObjects = {};
+        this.modelLoaded = false;
+        this.tempVec = vec3.create();
+    }
+
+    start() {
+        if (this.engine.xrSession) {
+            this.engine.xrSession.addEventListener(
+                'inputsourceschange',
+                this.onInputSourcesChange.bind(this)
+            );
+        } else {
+            this.engine.onXRSessionStart.add((session) => {
+                session.addEventListener(
+                    'inputsourceschange',
+                    this.onInputSourcesChange.bind(this)
+                );
+            });
+        }
+
+        this.handedness = ['left', 'right'][this.handednessIndex];
+    }
+
+    update() {
+        if (this.modelLoaded === true) {
+            this.mapGamepadInput();
+        }
+    }
+
+    onInputSourcesChange(event) {
+        event.added.forEach((xrInputSource) => {
+            const profile = this.customProfileFolder || xrInputSource.profiles[0];
+            /** load Profile json */
+            /** Note: if you are providing a custom window.ProfileJSON provide its path in the
+             * customProfileFolder property of the input-profile component.
+             */
+            this.url = this.path + profile + '/profile.json';
+            console.log('loading Profile ' + this.url);
+            fetch(this.url)
+                .then((res) => res.json())
+                .then((out) => {
+                    window.ProfileJSON = out;
```

Comment:
Please do not store stuff in window like that. If this can be configured by the user, just use a property on the object, or a shared property in the package that's exported